cmd/kube-apiserver/app/server.go

```diff
@@ -124,8 +124,27 @@ func Run(s *options.APIServer) error {
 		proxyDialerFn = tunneler.Dial
 	}
 
+	var proxyClientCerts []tls.Certificate
+	if s.KubeProxyClientCertFile != "" && s.KubeProxyClientKeyFile != "" {
+		clientCert, err := tls.LoadX509KeyPair(
+			s.KubeProxyClientCertFile,
+			s.KubeProxyClientKeyFile,
+		)
+
+		if err != nil {
+			glog.Fatalf("Invalid kube-proxy client certificates: %v", err)
+		} else {
+			glog.V(4).Infof("Using client certificate (%s, %s) when proxying over TLS",
+				s.KubeProxyClientCertFile, s.KubeProxyClientKeyFile)
+		}
+		proxyClientCerts = append(proxyClientCerts, clientCert)
+	}
+
 	// Proxying to pods and services is IP-based... don't expect to be able to verify the hostname
-	proxyTLSClientConfig := &tls.Config{InsecureSkipVerify: true}
+	proxyTLSClientConfig := &tls.Config{
+		InsecureSkipVerify: true,
```

Comment:
Not sure I follow what you mean. Are you saying that we should require services/pods to include a reference to a cluster or service IP? That might be difficult for services that use a public CA. On the other hand the proxy won't be able to verify private CAs or self-signed certificates, so that's why i was skipping verifying all together.

Note that the InsecureSkipVerify: true setting is already in master, all this PR adds is the ability to include client certs along with the request to the service or pod.